docassemble/ALToolbox/ThreePartsDate.py

```diff
@@ -112,24 +130,46 @@
       yearElement.on('change', updateDate);
       monthElement.on('change', updateDate);
       dayElement.on('change', updateDate);
-    });
-  });
+      idx += 1;
+    }});
+  }});
   
   // No jQuery validation, since it doesn't work on hidden elements
 """
 
 
+def check_empty_parts(item: str, default_msg="is not a valid date") -> Optional[str]:
+    parts = item.split("/")
+    empty_parts = [part == "" for part in parts]
+    if not any(empty_parts):
+        return None
+    if all(empty_parts):
+        return word("Enter a month, a day, and a year")
+    else:
+        empty_words = [
+            empty_part[1]
+            for empty_part in zip(empty_parts, ["month", "day", "year"])
+            if empty_part[0]
+        ]
+    if len(empty_words) == 2:
```

Comment:
This is probably fine but in gendered languages it's not simple enough to translate "a" the same for month, year and day. Might need separate sentences